dataland-frontend/src/components/forms/parts/elements/derived/ProcurementCategoryFormElement.vue

```diff
@@ -0,0 +1,177 @@
+              <div class="justify-content-between flex align-items-center">
+                <h5>{{ removeWordFromPhrase(el.value, el.label) }}</h5>
+                <div class="justify-content-end flex align-items-center">
+                  <FormKit
+                    type="number"
+                    :name="el.value"
+                    min="0"
+                    step="1"
+                    validation="required"
+                    validation-label="Number of suppliers per country"
+                    outer-class="my-0 mx-3"
+                  />
+                  <PrimeButton
+                    icon="pi pi-times"
+                    rounded
+                    class="p-button-icon"
+                    @click="removeItemFromListOfSelectedCountries(el.value)"
+                  />
+                </div>
+              </div>
+            </div>
+          </FormKit>
+        </div>
+      </div>
+    </FormKit>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import InputSwitch from "primevue/inputswitch";
+import { BaseFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import { FormKit } from "@formkit/vue";
+import UploadFormHeader from "@/components/forms/parts/elements/basic/UploadFormHeader.vue";
+import MultiSelectFormFieldBindData from "@/components/forms/parts/fields/MultiSelectFormFieldBindData.vue";
+import PrimeButton from "primevue/button";
+import NaceCodeFormFieldBindData from "@/components/forms/parts/fields/NaceCodeFormFieldBindData.vue";
+import { DropdownDatasetIdentifier, getDataset } from "@/utils/PremadeDropdownDatasets";
+import { LksgProcurementCategory } from "@clients/backend";
+
+export default defineComponent({
+  name: "ProcurementCategoryFormElement",
+  inject: {
+    procurementCategories: {
+      from: "procurementCategories",
+      default: {} as { [key: string]: LksgProcurementCategory },
+    },
+  },
+  components: {
+    InputSwitch,
+    FormKit,
+    UploadFormHeader,
+    MultiSelectFormFieldBindData,
+    PrimeButton,
+    NaceCodeFormFieldBindData,
+  },
+  props: BaseFormFieldProps,
+  data() {
+    return {
+      isItActive: !!this.procurementCategories[this.name],
+      procuredProductTypesAndServicesNaceCodesValue: [],
+      percentageOfTotalProcurementValue: "",
+      allCountry: getDataset(DropdownDatasetIdentifier.CountryCodes),
+      selectedCountries: [],
+      numberOfSuppliersPerCountryCodeValue: [],
+    };
+  },
+  watch: {
+    preSelectedCountries(newValue: []) {
+      this.selectedCountries = newValue;
+    },
+  },
+  computed: {
+    preSelectedCountries() {
+      return this.allCountry.filter((el) =>
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access,no-prototype-builtins
+        this.procurementCategories[this.name]?.numberOfSuppliersPerCountryCode?.hasOwnProperty(el.value)
+      );
+    },
+  },
```

Comment:
This sure works, but seems a little overkill to me:

Why not checking for preSelectedCountries only once when mounting the component?
After that it is not necessary anymore to keep track of changes.

Reply:
done

Reply:
👍